### PR TITLE
Avoid additional uncached requests to SegmentEditor.getAll

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

With https://github.com/matomo-org/matomo/pull/21521 we introduced a check within the API method `SegmentEditor.getAll` to filter away segments that aren't valid anymore. Depending on the amount of stored segment, this might take some time.

To avoid the filtering multiple times during a request we should avoid calling the API directly, but use `SegmentEditor::getAllSegmentsForSite` instead. This method caches the results for the current request.

This PR changes a couple of direct API usages to use `SegmentEditor::getAllSegmentsForSite` instead.

refs DEV-17463

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreVisualizations/JqplotDataGenerator.php [hunk 1]
```diff
@@ -78,8 +77,6 @@ public function __construct($properties, $graphType, JqplotGraph $graph)
         $this->graphType = $graphType;
         $this->isComparing = $graph->isComparing();
         $this->graph = $graph;
-
-        $this->availableSegments = Request::processRequest('SegmentEditor.getAll', $override = [], $default = []);
```
Comment (sgiehl) [inline plugins/CoreVisualizations/JqplotDataGenerator.php:82]: Note: This property seems unused in this class and as it is declared `private` it wouldn't be usable in inherited classes, so should be safe to simply remove it.

File: plugins/SegmentEditor/SegmentSelectorControl.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function __construct($idSite = false)
         $this->nameOfCurrentSegment = '';
         $this->isSegmentNotAppliedBecauseBrowserArchivingIsDisabled = 0;
 
-        $this->availableSegments = Request::processRequest("SegmentEditor.getAll", ['idSite' => $this->idSite], $defaultRequest = []);
+        $this->availableSegments = SegmentEditor::getAllSegmentsForSite($this->idSite);
```
Comment (mneudert) [inline plugins/SegmentEditor/SegmentSelectorControl.php:71]: This change reorders the segments in the frontend matching the [UI test failure](https://builds-artifacts.matomo.org/matomo-org/matomo/dev-17463/6944483032/SegmentSelectorEditorTest_updated.png).

Is that an intended/good change or should the sorting in `getAllSegmentsForSite` be adjusted? Or just sort again to match UI expectations?

File: plugins/ScheduledReports/Controller.php [hunk 3]
```diff
@@ -116,8 +117,8 @@ public function index()
             $savedSegmentsById = array(
                 '' => Piwik::translate('SegmentEditor_DefaultAllVisits')
             );
-            $response = Request::processRequest("SegmentEditor.getAll", ['idSite' => $this->idSite], $defaultRequest = []);
-            foreach ($response as $savedSegment) {
+            $allSegments = SegmentEditor::getAllSegmentsForSite($this->idSite);
```
Comment (michalkleiner) [inline plugins/ScheduledReports/Controller.php:120]: Does this couple the plugins too tightly and could've been the reason it was previously done via an API call?
Does it remove any extensibility here possibly if someone manipulates the API request/response in their plugins?

Comment (sgiehl) [reply plugins/ScheduledReports/Controller.php:120]: `SegmentEditor::getAllSegmentsForSite` also uses the same API to fetch the data.

File: plugins/SegmentEditor/SegmentSelectorControl.php (hunk 2 above)
Comment (sgiehl) [reply plugins/SegmentEditor/SegmentSelectorControl.php:71]: I've just checked the places again where that method actually was used before and actually nowhere did the order matter. So I'll simply remove the reordering in `SegmentEditor::getAllSegmentsForSite`

**Review by mneudert — APPROVED** [review]

**Pull request merged**